ops/check-changed/main.py

```diff
@@ -0,0 +1,126 @@
+import logging.config
+import os
+import re
+import subprocess
+import sys
+
+from github import Github
+
+WHITELISTED_BRANCHES = {
+    'master',
+    'develop'
+}
+
+LOGGING_CONFIG = {
+    'version': 1,
+    'disable_existing_loggers': True,
+    'formatters': {
+        'standard': {
+            'format': '%(asctime)s [%(levelname)s]: %(message)s'
+        },
+    },
+    'handlers': {
+        'default': {
+            'level': 'INFO',
+            'formatter': 'standard',
+            'class': 'logging.StreamHandler',
+            'stream': 'ext://sys.stderr'
+        },
+    },
+    'loggers': {
+        '': {
+            'handlers': ['default'],
+            'level': 'INFO',
+            'propagate': False
+        },
+    }
+}
+
+logging.config.dictConfig(LOGGING_CONFIG)
+log = logging.getLogger(__name__)
+
+
+def main():
+    patterns = sys.argv[1].split(',')
+    patterns.append(r'^\.circleci/\.*')
+
+    fp = os.path.realpath(__file__)
+    monorepo_path = os.path.realpath(os.path.join(fp, '..', '..'))
+
+    log.info('Discovered monorepo path: %s', monorepo_path)
+    current_branch = git_cmd('rev-parse --abbrev-ref HEAD', monorepo_path)
+    log.info('Current branch: %s', current_branch)
+
+    if current_branch in WHITELISTED_BRANCHES:
+        log.info('Current branch %s is whitelisted, triggering build', current_branch)
+        exit_build()
+
+    pr_urls = os.getenv('CIRCLE_PULL_REQUESTS', None)
+    pr_urls = pr_urls.split(',') if pr_urls else []
+    if len(pr_urls) == 0:
+        log.info('Not a PR build, skipping build')
+        exit_nobuild()
+    if len(pr_urls) > 1:
+        log.warning('Multiple PR URLs found, choosing the first one. PRs found:')
+        for url in pr_urls:
+            log.warning(url)
+
+    gh_token = os.getenv('GITHUB_ACCESS_TOKEN')
+    if gh_token is None:
+        log.info('No GitHub access token found - likely a fork. Triggering build')
+        exit_build()
+
+    g = Github(gh_token)
+    try:
+        g.get_user()
+        repo = g.get_repo(os.getenv('CIRCLE_PROJECT_USERNAME') + '/' + os.getenv('CIRCLE_PROJECT_REPONAME'))
+    except Exception:
+        log.exception('Failed to get repo from GitHub')
+        exit_build()
+
+    pr = repo.get_pull(int(pr_urls[0].split('/')[-1]))
```

Comment:
Could `pr` ever be `None` here? resulting in `pr.{base,head}` not existing? Or if `repo.get_pull` throws if it doesn't exist?

Reply:
Ah i see the exception handler at the bottom that would catch this